Create release_template.md (#249)
enthought/apptools: .github/ISSUE_TEMPLATE/release_template.md

---

> name: Release Checklist

Reviewer: When I opened an issue for pip GitHub repo, I noticed their template title and description make it very obvious when a template is designed for maintainers only.  I think it is a good idea so that the majority of users who are not maintainers can ignore the template quickly.

e.g. Change name from "Release Checklist" -> "(Maintainers only) Release Checklist"

about: Use this template for upcoming releases.
title: "[Release Number] Release Checklist"
---

Target date:

Release Candidate by YYYY-MM-DD

Release by YYYY-MM-DD

Release candidate
-------------------
- [ ] Verify that all relevant PRs have been merged to master.
- [ ] Create a PR against master to bump version number, merge that PR
- [ ] From the commit just before bumping the version, create a new branch `maint/<release version number>`

> - [ ] Update changelog and open PR targeting a new `maint/<release version number>` branch

Reviewer: This is where we might want to mention the `etstool.py changelog build` command? There is a mention in https://github.com/enthought/apptools/blob/4f4178c86ca0393555715245b6ac5047b1434cad/etstool.py#L73-L80 but it is a bit hidden. The release checklist might be a better place for at least a mention. We probably want to make sure the used news fragment are deleted in this step? If so, let's mention it too.

(For later, we may revisit if we want to update the changelog build command such that the clean action is done by default rather than optionally by prompt.)

Release
-------
- [ ] Create branch `release/<release version number>` from `maint/<release version number>` branch.
  - [ ] Set release to `<release version number>`, and set `IS_RELEASED` is true; commit
  - [ ] Install from source distribution and run tests again
  - [ ] Open a PR against `maint/<release version number>` with this being the last commit so that CI is built on the release commit
  - [ ] Once CI is done building merge PR
  - [ ] Bump the micro version number i.e. `<release version number + 0.0.1>` and set `IS_RELEASED` to false; commit.
  - [ ] Open and then merge a separate PR against `maint/<release version number>`
- [ ] From the commit at which `IS_RELEASED` is true and version is `<release version number>`, tag (annotated!) `git tag -a -m "Release <release version number>" <release version number> <commit-hash>`
- [ ] Push the tag `git push origin <release version number>`
- [ ] Make PR targeting `gh-pages` branch: Generate documentation and copy the content to the branch. Verify that the resulting index.html looks good.
- [ ] Upload to PyPI
- [ ] Test the PyPI package

Post-release
-------------
- [ ] Package update for `enthought/free` repository (for EDM)

> - [ ] Backport release note and change log to master, and possibly `maint/<release version number>` branch.

Reviewer: ```suggestion
- [ ] Backport change log (and any deletion of news fragments) to master 
```

> - [ ] Update GitHub Release pages https://github.com/enthought/envisage/releases

Reviewer: ```suggestion
- [ ] Update GitHub Release pages https://github.com/enthought/apptools/releases
```
